### PR TITLE
Feature/narwhals weights mixin

### PR DESCRIPTION
**Pull request by limlam96**
https://github.com/lvgig/tubular/issues/322

### CODE REVIEW

File: tubular/mixins.py [hunk 1]
```diff
@@ -187,13 +140,14 @@ class attribute, indicates whether transformer has been converted to polars/pand
...
-    def check_weights_column(self, X: pd.DataFrame, weights_column: str) -> None:
+    @nw.narwhalify
+    def check_weights_column(self, X: FrameT, weights_column: str) -> None:
         """Helper method for validating weights column in dataframe.
 
         Args:
         ----
-            X (pd.DataFrame): df containing weight column
-            weights_column (str): name of weight column
+            dfs: df containing weight column
```
Comment (davidhopkinson26) [inline tubular/mixins.py:149]: shouldnt this be X? realise we didn't update docstrings in last PR to reflect that X can be a pd or pl Dataframe.

**Review by davidhopkinson26 — CHANGES_REQUESTED** [review]

File: tubular/mixins.py (hunk 1 above)
Comment (limlam96) [reply tubular/mixins.py]: ah thanks, updated here https://github.com/lvgig/tubular/pull/331/commits/161db20158b881dc1d23b4f5703422cf0e30d6a4

File: tests/base_tests.py [hunk 2]
```diff
@@ -511,24 +542,46 @@ def test_fit_not_changing_data_weighted(
...
+        uninitialized_transformer = uninitialized_transformers[self.transformer_name]
+
+        # skip polars test if not narwhalified
+        if not uninitialized_transformer.polars_compatible and isinstance(
+            df,
+            pl.DataFrame,
+        ):
+            return
+
+        df = nw.from_native(df)
```
Comment (davidhopkinson26) [inline tests/base_tests.py:554]: is this to get around the issue you were having with narwhalified methods in super calls?

**Review by davidhopkinson26 — APPROVED** [review]
thanks @limlam96!

**Pull request merged**